Packet++/src/TcpLayer.cpp

```diff
@@ -44,6 +45,20 @@ TcpOptionBuilder::TcpOptionBuilder(NopEolOptionTypes optionType)
 	}
 }
 
+TcpOptionBuilder::TcpOptionBuilder(const NopEolOptionEnumType optionType)
+{
+	switch (optionType)
+	{
+	case NopEolOptionEnumType::Eol:
+		init(static_cast<uint8_t>(PCPP_TCPOPT_EOL), nullptr, 0);
+		break;
+	case NopEolOptionEnumType::Nop:
+	default:
+		init(static_cast<uint8_t>(PCPP_TCPOPT_NOP), nullptr, 0);
+		break;
+	}
```

Comment:
Shouldn't we use `TcpOptionEnumType::Eol` and `TcpOptionEnumType::Nop` instead of `PCPP_TCPOPT_EOL` and `PCPP_TCPOPT_NOP` here?

Reply:
Btw. I have a question: what is the difference between: `TcpOptionEnumType::Eol` and `NopEolOptionEnumType::Eol` ? 

Can they be used interchangeably? Because `TcpOptionEnumType::Eol == 0` but `NopEolOptionEnumType::Eol == 1`

Reply:
Yes, they are more or less the same. The reason it's a different enum is because this `TcpOptionBuilder` c'tor is specifically for creating NOP or EOL TCP options which are different from other options in their size, as mentioned in the doxygen documentation. If we let the user use any `TcpOptionEnumType` we may get values that are not NOP or EOL

Reply:
This enum is only used in the c'tor of `TcpOptionBuilder`, it's another way to say "this c'tor can only accept EOL or NOP". Everywhere else the `TcpOptionEnumType` enum should be used. I'm not sure what can be fixed here... 🤔